cmd/kubeadm/app/cmd/phases/workflow/runner.go

```diff
@@ -442,8 +442,11 @@ func (e *Runner) prepareForExecution() {
 // addPhaseRunner adds the phaseRunner for a given phase to the phaseRunners list
 func addPhaseRunner(e *Runner, parentRunner *phaseRunner, phase Phase) {
 	// computes contextual information derived by the workflow managed by the Runner.
-	generatedName := strings.ToLower(phase.Name)
-	use := generatedName
+	use := strings.ToLower(phase.Name)
+	if pos := strings.Index(use, " "); pos != -1 {
+		use = use[:pos]
```

Comment:
if we don't include the `[api-server-endpoint]` as part of the phase names we can avoid the hack. :(
would it suffice to have the `[api-server-endpoint]` as part of `Example:`?

Reply:
Maybe we could add a specific field to `Phase` struct to make this logic cleaner?

Reply:
I don' think we should overengineer this.
having args descriptor as part of the name is usual in cobra (nb phases `Name` maps to cobra `Use` attribute) , and it makes a nice "Usage" (prior art e.g. kubectl). in this case
```
Usage:
  kubeadm join [api-server-endpoint] [flags]
```
The logic above, that now I have refactored in a separated `cleanName` function and documented, starts from what we are passing to cobra and returns a cleaned version that we are going to use in the list of phases

Reply:
ok